### PR TITLE
Fix #4087 - Created a proxy object to hide JS properties.

### LINKED ISSUE #4087: JS leaks private properties via window.__firefox__
### Description: 
- Going to any page and typing `window.__firefox__` will leak private properties so that if someone does `Object.keys(window.__firefox__)` it will NOT show the properties, but enumerating it with `Object.getOwnPropertyNames(window.__firefox__)` WILL display a list of all properties and they can be enumerated & changed by malicious JS.

**Actual result:** 
Properties are enumerable and they leak to the page.

**Expected result:**
Should not leak properties. They should NOT be enumerable.

**Reproduces how often:** [Easily reproduced, Intermittent Issue]
Easily reproduced

**Brave Version:** 
Any

### PR DESCRIPTION
**Pull request by Brandon-T**
Properties can only be hidden if they are non-configurable & writable.
If a property is non-configurable, readonly, it is impossible to hide as a TypeError will be thrown.

## Summary of Changes
- Hides JS properties like those added `window.__firefox__`.
- The properties can only be retrieved in code if you know the EXACT name of it. Since we use the `securityToken` everywhere (hopefully), all properties should be hidden and inaccessible (aka WKContentWorld for below iOS 14), but also in cases where the page must have access to the JS where ContentWorld sandbox can't be used, this is the alternative :)

This pull request fixes #4087

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Test Plan:
N/A

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

**Comment by Brandon-T** [comment]
Without this: 
image

With this:
image

By returning `[]` from `ownKeys` function, we get:
image

Making it impossible to enumerate the properties. Alternatively, instead of returning `[]` we can just make all the properties `configurable: true` OR `enumerable: true` and they will be hidden automatically just like `userScripts` and `BSHE7BF....` (BraveSearch) as shown in the first screenshot. Doing so will result in `Object.getOwnPropertyNames(window.__firefox__)` returning `[]`

**Review by iccub — APPROVED** [review]
Needs a test plan, we can sanity check that adding items to playlist works, does it affect other js too? like reader mode

File: Client/Frontend/UserContent/UserScripts/AllFrames/AtDocumentEnd/__firefox__.js [hunk 1]
```diff
@@ -10,19 +10,82 @@ if (!window.__firefox__) {
...
-      userScripts: {},
-      includeOnce: function(userScript, initializer) {
-        if (!__firefox__.userScripts[userScript]) {
-          __firefox__.userScripts[userScript] = true;
-          if (typeof initializer === 'function') {
-            initializer();
-          }
-          return false;
-        }
+    value: (function() {
```
Comment (jumde) [inline Client/Frontend/UserContent/UserScripts/AllFrames/AtDocumentEnd/__firefox__.js:13]: nit: I'm not sure I understand why we need `__firefox__.js` added at both `AtDocumentStart` and `AtDocumentEnd` but we can file a follow-up issue to figure this out.

**Review by jumde — APPROVED** [review]
lgtm

**Pull request merged**